### PR TITLE
[SE-3173] Updates XBlock to Support Python 3 for Juniper Release

### PR DESCRIPTION
**Pull request by nizarmah**
Updates XBlock to support Python 3 so that it is compatible with edX's Juniper release.

Changes include PR #84 :+1: 

**JIRA tickets**: SE-3173

**Sandbox URL**: 
- LMS : https://xblock-image-explorer-test.opencraft.hosting/
- Studio: https://studio.xblock-image-explorer-test.opencraft.hosting/

**Testing instructions**:

1. Login to the LMS with `staff`
2. Open the Studio and make sure that you can add an Image XBlock to a Course
3. Make sure that the XBlock renders properly in Studio
4. Press _View Live_ and make sure that you can interact with the XBlock on the LMS 
5. Press _Preview_ and make sure that you can preview the XBlock properly

**Reviewers**
- [x] @toxinu
- [x] @shafqatfarhan 
- [x] @ihtram

### CODE REVIEW

**Comment by toxinu** [comment]
@nizarmah Looks good to merge 👍🏻 

cc @shafqatfarhan @musmanmalik @msaqib52 Would you have to review this pull request please?

**Comment by shafqatfarhan** [comment]
@nizarmah I already have a similar [PR](https://github.com/edx-solutions/xblock-image-explorer/pull/81) (Python 3 upgrade work for this repo) in my queue that is under review. It seems like both the PRs are targeting Python 3 upgrade work. I will get back to you once I am done reviewing the other PR

cc: @ihtram

**Comment by nizarmah** [comment]
Hello @shafqatfarhan ! 
First of all, thank you for your time.

I noticed the other PR, but it has a Merge Conflict. And I wasn't a fan of how some of the changes were made.

It seems like the other PR keeps support for Python 2.7, while supporting Python 3.

This PR removes support for Python 2.7, similarly to what edX has done in many of their repositories lately.

In case you choose to merge this PR, all `tox` tests work, there are no merge conflicts, and the explorer works.

**Review by shafqatfarhan — CHANGES_REQUESTED** [review]
Thank you for the PR. I totally get your point that the other similar PR supports Python 2.7 as well and it seems like we don't need to support that now.

I have reviewed your PR and added a few suggestions. Other than that, please make sure:
- to squash the commits
- CircleCI tests pass (CI tests did not run on your PR)
- to change the base branch to `juniper-rebase` since the master branch is being used in active development so, we will be using `juniper-rebase` for Juniper release. 

Thanks

**Comment by shafqatfarhan** [comment]
@nizarmah Are you planning to work on other repos of edx-solutions organization on PY3 upgrade work? Just curious, because my team has also planned to work on the same upgrade work for edx-solutions organization as part of Juniper rebase for edx-solutions/edx-platform.

File: .circleci/config.yml
Comment (nizarmah) [reply .circleci/config.yml:5]: Yeah, there shouldn't be any problem with python 3.5 :+1: I'll definitely update that.

**Comment by nizarmah** [comment]
Hello @shafqatfarhan. I'm not really sure exactly if we do have other XBlocks we want to upgrade to support Juniper. However, of what I am aware of, we worked on most upgrades, and Image Explorer turned out to be an exception that we forgot.

Would you please be able to share with me a list of the XBlocks you plan to upgrade so that I can inform my peers at OpenCraft about them?

**Comment by shafqatfarhan** [comment]
> Hello @shafqatfarhan. I'm not really sure exactly if we do have other XBlocks we want to upgrade to support Juniper. However, of what I am aware of, we worked on most upgrades, and Image Explorer turned out to be an exception that we forgot.
> 
> Would you please be able to share with me a list of the XBlocks you plan to upgrade so that I can inform my peers at OpenCraft about them?

Ideally, we have to upgrade all the XBlocks / extension apps to PY3 that are in edx-solutions organization. If you plan to work on any other XBlock / extension app under this organization, let us know so that there won't be duplication of work.

Thanks

**Review by shafqatfarhan — APPROVED** [review]
👍

**Comment by nizarmah** [comment]
> If you plan to work on any other XBlock / extension app under this organization, let us know so that there won't be duplication of work.

Will do :slightly_smiling_face: Thanks a lot for your review @shafqatfarhan! :heart: 
Is someone else's review needed before this can be merged?

**Review by ihtram — APPROVED** [review]
LGTM. Please go ahead and merge this PR

**Comment by nizarmah** [comment]
@shafqatfarhan can you please merge this when you have time? I don't have the necessary permissions

**Comment by ihtram** [comment]
@nizarmah merging this PR. For now we will using `juniper-rebase` branch for this Xblock. Once we will be ready with juniper upgrade, we will merge this change into master and will then create a release tag.

**Pull request merged**
